docs/proposals/OTA(over-the-air)-upgrades-for-edge-node.md

```diff
@@ -0,0 +1,154 @@
+
+# OTA(Over-The-Air) Upgrades For Edge Node
+
+## Motivation
+
+In order to make the edge node more convenient and rapid upgrade, we introduce a remote upgrade scheme OTA (Over-The-Air) into KubeEdge.In the main process of OTA(i.e. make the bundle, download the bundle, verify the bundle and firmware upgrade), we have realized most steps.Our release will generate a new image version called installation-package, then we use the NodeUpgradeJob CRD to obtain the installation tool keadm in the image and run the command to upgrade the edge node.During this process, if the hacker masquerades the image in the edge node, this will result in the untrusted binary keadm. We need to verify the digest of the image before the keadm executes the upgrade, which is the third step of OTA to verify the bundle.And in some business scenarios (Internet of vehicles, Internet of Things), we also need to provide an option to make the node wait for confirmation from a person with permission before upgrading the edge node.
+
+## Goals
+
+- Update the proposal of the NodeUpgradeJob
+- During the edge node upgrade, the keadm can verify the image digest. If the image is invalid, the edge node cannot be upgraded and an error message is reported
+- If the edge node upgrade confirmation is enabled, the upgrade cannot be executes before the confirmation and the job will always be waiting
+- Users can call the metaserver API or execute command `keadm ctl ...` to confirm the upgrade
+
+## Project framework diagram
+
+![](../images/proposals/over-the-air-upgrades-for-edge-node.png)
+
+## Implementation details
+### Validation of the Image Digest Before Edge Node Upgrade
+
+#### Objective
+Ensure that the image being used for the upgrade is authentic and has not been tampered with.
+#### Steps
+- Get image digest(**Ensure Docker Daemon is Running**):When creating an installation package image, obtain the cryptographic hash (digest) of the image through docker's API.Docker provides an API that allows you to interact with the Docker daemon programmatically.*For example*:Use `curl` to interact with the Docker registry API. Replace your_registry, your_image, and tag with your actual values
+```bash
+curl -X GET -H "Accept: application/vnd.docker.distribution.manifest.v2+json"
+\
+https://your_registry/v2/your_image/manifests/tag
+```
+Response:
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+  "config": {
+    "mediaType": "application/vnd.docker.container.image.v1+json",
+    "size": 7023,
+    "digest": "sha256:abc123..."
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+      "size": 32654,
+      "digest": "sha256:def456..."
+    }
+  ]
+}
+```
+- Release image digest:Pass the obtained image digest to the edge using Hub by calling this `commontypes.NodeUpgradeJobRequest` method.
+```go
+func (ndc *NodeUpgradeController) processUpgrade(upgrade *v1alpha1.NodeUpgradeJob) {
```

Comment:
Just describe the key points by pseudocode. Use `klog.logger` instead of `fmt.Print` to output information.